Changed the use of pyglet plot module

tokencolour opened the pull request:
This example uses, as of now, the old pyglet plotting module
which is not working right now.
Changed so that it uses the new plot API.
Trying to fix #10243

tokencolour committed:
Changed the use of pyglet plot module

This example uses, as of now, the old pyglet plotting module
which is not working right now.
Changed so that it uses the new plot API.
Trying to fix #10243

moorepants commented on examples/advanced/autowrap_ufuncify.py:
```diff
@@ -46,7 +47,7 @@ def main():
 
     # Let's also plot the ufunc's we generate
-    plot1 = Plot(visible=False)
+    #plot1 = Plot(visible=False)
```
Don't leave in old commented code.

tokencolour replied:
I'll change that.

moorepants commented:
It seems to me you completely removed the purpose of the example. I think the original author was trying to show how to make use of a binary_ufunc in plotting to speed things up. The new plot command is simply plotting the sympy expression. If the new plot command is using a fast numerical evaluation in the background, then maybe this example isn't useful anymore. But my guess is that the plot command uses evalf, so passing in the generated function would make the plotting faster.

tokencolour commented:
The problem is that I am not able to pass a sympified expression to `plot` method once it is ufuncified. Passing `expr` directly defeats the purpose, but I had no choice. Could you please suggest something?

moorepants commented:
That sounds like `plot()` may need more than the `evalf` method then. Theorectically, this should work:

```python
expr = x**2
fast_expr = binary_ufunc(expr, x)
plot(fast_expr)
```

tokencolour commented:
By `binary_ufunc` do you mean the compiled  unevaluated ufunc created by ufuncifying the expr?:
as in 
```
>>> binary_ufunc=ufuncify(x, expr)
>>> fast_expr= binary_ufunc(expr, x)
```

tokencolour commented:
If that is true, that is where my problem is, binary_ufunc expects arrays. I wish there was a way we could get back to the expression once we ufuncified it.

moorepants commented:
I don't think you want to get the expression back. You need to have a look at how the `plot` function works and figure out why the generated function doesn't work.

moorepants commented:
Actually, I think you should just use matplotlib's plot function with the ufuncified expression.

The `binary_function()` creates a SymPy function, not a SymPy expression.

tokencolour commented:
I'll have to see if matplotlib's plot function will be able to handle a Sympy expression. Plotting a ufuncified function is no help with Sympy's plot.  I had tried this:
```
 g=implemented_function('g', binary_poly)
 plot(g(x))
```
I get a TypeError. Kindly look at https://github.com/sympy/sympy/issues/10243#issuecomment-166097361

moorepants commented:
You don't want to use a sympy expression, just do:

```python
expr = x**2
fast_expr = ufuncify(x, expr)
import matplotlib.pyplot as plt
plt.plot(range(10), fast_expr(range(10)))
plt.show()
```

tokencolour commented:
This is working fine! I'll add this. Thank you!

tokencolour committed:
Changed the use of pyglet Plot module

This example uses, as of now, the old pyglet plotting module
which is not working right now.
My fix uses matplotlib as suggested by @moorepants.
Fixes #10243

tokencolour commented:
@moorepants Kindly take a look at this.

moorepants commented on examples/advanced/autowrap_ufuncify.py:
```diff
@@ -72,11 +72,11 @@ def main():
 
 if __name__ == '__main__':
-    main()
+    main()
```
Missing end of line.

moorepants commented on examples/advanced/autowrap_ufuncify.py:
```diff
@@ -26,9 +26,10 @@
     sys.exit("Cannot import numpy. Exiting.")
 
 import mpmath
+import matplotlib.pyplot as plt
```
Since matplotlib is not a sympy dependency, you should do the same thing as numpy above.

tokencolour committed:
Changed the use of pyglet Plot module

Added EOL after main and proper importing of matplotlib.
Fixes #10243

tokencolour commented:
Kindly take a look at this.

moorepants commented:
Looks good!

tokencolour commented:
Thank you!
